go-controller/pkg/cni/helper_linux.go

```diff
@@ -818,3 +827,64 @@ func (pr *PodRequest) deletePorts(ifaces []string, podNamespace, podName string)
 		pr.deletePort(iface, podNamespace, podName)
 	}
 }
+
+// setupIngressFilter sets up an ingress filter using nftables to block
+// unwanted ICMPv6 Router Advertisement (RA) packets on a specific device.
+// It creates a new nftables table, chain, and rule to drop RA packets
+// that do not match the specified gateway link-local address (gwLLA) and
+// have a Router Advertisement (RA) lifetime not equal to 0.
+//
+// The nftables rule created by this function looks like:
+// `icmpv6 type nd-router-advert ip6 saddr != <gwLLA> @th,48,16 != 0 drop`
+//
+// Parameters:
+// - gwLLA: The gateway link-local address to match against in the RA packets.
+//
+// Returns:
+// - error: An error if the nftables setup fails, otherwise nil.
+func setupIngressFilter(gwLLA string) error {
+	const (
+		tableName   = "ingress_filter"
+		rasLifetime = "@th,48,16" // Offset for RA lifetime field in ICMPv6 packets
+	)
+
+	nft, err := knftables.New(knftables.NetDevFamily, tableName)
+	if err != nil {
+		return fmt.Errorf("failed to initialize table: %w", err)
+	}
+	return setupIngressFilterWithTableAndLifetimeMatcher(nft, gwLLA, rasLifetime)
+}
+
+func setupIngressFilterWithTableAndLifetimeMatcher(nft knftables.Interface, gwLLA, rasLifetime string) error {
+	const (
+		chainName  = "input"
+		deviceName = "ovn-udn1"
```

Comment:
NIT: This should be a constant.
Currently it is specified here https://github.com/ovn-kubernetes/ovn-kubernetes/blob/1a5a6c27856d59fc31ac9b5c391b8d0db055701d/go-controller/pkg/cni/udn/primary_network.go#L29 but could be extracted so it can be reused. 
This can be a followup.

Reply:
well this is called from `setupInterface` so clearly we should pass tha interface to this filter function, fixing...